compiler/ballerina-lang/src/main/java/io/ballerina/projects/DocumentContext.java

```diff
@@ -139,19 +140,45 @@ private Set<ModuleLoadRequest> getModuleLoadRequests(ModuleDescriptor currentMod
+    }
+
+    private static void addTransactionModuleImportIfRequired(ModuleDescriptor currentModuleDesc,
+                                                             PackageDependencyScope scope,
+                                                             ModulePartNode modulePartNode,
+                                                             Set<ModuleLoadRequest> moduleLoadRequestSet) {
         // TODO This is a temporary solution for SLP6 release
         // TODO Traverse the syntax tree to see whether to import the ballerinai/transaction package or not
         TransactionImportValidator trxImportValidator = new TransactionImportValidator();
+        if (!trxImportValidator.shouldImportTransactionPackage(modulePartNode)) {
+            return;
+        }
+        addModuleLoadRequest(currentModuleDesc, scope, moduleLoadRequestSet, Names.BALLERINA_INTERNAL_ORG.value,
+                Names.TRANSACTION.value, DependencyResolutionType.PLATFORM_PROVIDED);
+    }
 
-        if (trxImportValidator.shouldImportTransactionPackage(modulePartNode) &&
-                !currentModuleDesc.name().toString().equals(Names.TRANSACTION.value)) {
-            String moduleName = Names.TRANSACTION.value;
-            ModuleLoadRequest ballerinaiLoadReq = new ModuleLoadRequest(
-                    PackageOrg.from(Names.BALLERINA_INTERNAL_ORG.value),
-                    moduleName, scope, DependencyResolutionType.PLATFORM_PROVIDED);
-            moduleLoadRequestSet.add(ballerinaiLoadReq);
+    private static void addNaturalProgrammingModuleImportIfRequired(ModuleDescriptor currentModuleDesc,
+                                                                    PackageDependencyScope scope,
+                                                                    ModulePartNode modulePartNode,
+                                                                    Set<ModuleLoadRequest> moduleLoadRequestSet) {
+        NaturalProgrammingImportAnalyzer naturalProgrammingImportAnalyzer = new NaturalProgrammingImportAnalyzer();
+        if (!naturalProgrammingImportAnalyzer.shouldImportNaturalProgrammingModule(modulePartNode)) {
+            return;
+        }
+        addModuleLoadRequest(currentModuleDesc, scope, moduleLoadRequestSet, Names.BALLERINA_ORG.value,
+                Names.NATURAL_PROGRAMMING.value, DependencyResolutionType.PLATFORM_PROVIDED);
+    }
+
+    private static void addModuleLoadRequest(ModuleDescriptor currentModuleDesc, PackageDependencyScope scope,
+                                             Set<ModuleLoadRequest> moduleLoadRequestSet, String orgName,
+                                             String moduleName, DependencyResolutionType dependencyResolutionType) {
+        if (!currentModuleDesc.name().toString().equals(moduleName)) {
+            ModuleLoadRequest moduleLoadRequest = new ModuleLoadRequest(
+                    PackageOrg.from(orgName), moduleName, scope, dependencyResolutionType);
```

Comment:
In here Don't we need to consider about the language version
As an example np module can have different versions for U12 and U13

Reply:
That's resolved by the project API? Which is why we shouldn't specify a version here?